src/libs/OptionsListUtils.ts

```diff
@@ -1625,131 +1669,142 @@ function getValidReports(reports: OptionList['reports'], config: GetValidReports
         includeSelfDM = false,
         includeInvoiceRooms = false,
         action,
-        selectedOptions = [],
         includeP2P = true,
         includeDomainEmail = false,
-        shouldBoldTitleByDefault = true,
         loginsToExclude = {},
-        shouldSeparateSelfDMChat,
-        shouldSeparateWorkspaceChat,
         excludeNonAdminWorkspaces,
-        isPerDiemRequest = false,
-        showRBR = true,
     } = config;
     const topmostReportId = Navigation.getTopmostReportId();
 
-    const validReportOptions: OptionData[] = [];
-    const workspaceChats: OptionData[] = [];
-    let selfDMChat: OptionData | undefined;
-    const preferRecentExpenseReports = action === CONST.IOU.ACTION.CREATE;
+    // eslint-disable-next-line rulesdir/prefer-at
+    const option = reportOption;
+    const report = reportOption.item;
+    const chatReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${option.item.chatReportID}`];
+    const doesReportHaveViolations = shouldDisplayViolationsRBRInLHN(report, transactionViolations);
 
-    for (let i = 0; i < reports.length; i++) {
-        // eslint-disable-next-line rulesdir/prefer-at
-        const option = reports[i];
-        const report = option.item;
-        const chatReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${report.chatReportID}`];
-        const doesReportHaveViolations = shouldDisplayViolationsRBRInLHN(report, transactionViolations);
+    const shouldBeInOptionList = shouldReportBeInOptionList({
+        report,
+        chatReport,
+        currentReportId: topmostReportId,
+        betas,
+        doesReportHaveViolations,
+        isInFocusMode: false,
+        excludeEmptyChats: false,
+        includeSelfDM,
+        login: option.login,
+        includeDomainEmail,
+        isReportArchived: !!option.private_isArchived,
+    });
 
-        const shouldBeInOptionList = shouldReportBeInOptionList({
-            report,
-            chatReport,
-            currentReportId: topmostReportId,
-            betas,
-            doesReportHaveViolations,
-            isInFocusMode: false,
-            excludeEmptyChats: false,
-            includeSelfDM,
-            login: option.login,
-            includeDomainEmail,
-            isReportArchived: !!option.private_isArchived,
-        });
+    if (!shouldBeInOptionList) {
+        return false;
+    }
 
-        if (!shouldBeInOptionList) {
-            continue;
-        }
+    const isThread = option.isThread;
```

Comment:
Not sure all these local variables are necessary either. It probably doesn't matter since babel is probably optimizing them away at compile time anyways, but I'm not sure they're helping readability meaningfully anyways